server/src/main/java/org/elasticsearch/lucene/queries/BlendedTermQuery.java

```diff
@@ -148,7 +148,10 @@ protected int compare(int i, int j) {
             if (prev > current) {
                 actualDf++;
             }
-            contexts[i] = ctx = adjustDF(reader.getContext(), ctx, Math.min(maxDoc, actualDf));
+
+            int docCount = reader.getDocCount(terms[i].field());
```

Comment:
This query's score calculation is complex and can be hard to reason about. I liked this change because it doesn't affect the non-broken case at all. It only changes the score if `docCount < docFreq`, which would have resulted in a negative IDF and a broken score anyways.